### ISSUE TITLE
Checkout block integration broken in latest release

### ISSUE THREAD

**Issue by dechov** [Automattic/woocommerce-payments#1284]
As of this week, the payment method is **not showing up at all in the Checkout block**, and a console error is thrown. There appear to be two _independent_ reasons for this:

- With WooCommerce Payments 1.9.0 (released Feb 2) or later, a JS error is thrown: `Uncaught ReferenceError: wcpay_config is not defined` – not sure exactly what could be causing it [[full changeset](https://github.com/Automattic/woocommerce-payments/compare/1.8.0...1.9.0)], but at a glance, I'm not entirely sure how it is that we're depending on that global in the blocks script to begin with – do we need to localize the script [here](https://github.com/Automattic/woocommerce-payments/blob/cf3c9e3aa1fb9abce0c83414afe29168ed16e697/includes/class-wc-payments-blocks-payment-method.php#L31-L46)? (According to [this doc](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/extensibility/payment-method-integration.md#server-side-integration), it looks like maybe we should be accessing that data differently: `wc.wcSettings.getSetting( 'woocommerce_payments_data' )`?)

- With WooCommerce Payments 1.8.0 or earlier but with WooCommerce Blocks 4.4.0 (released Feb 3), a different JS error is thrown: `Uncaught Error: The features property for the payment method must be an array.` – this is because the payment method requires a new `features` array (inside a `supports` object property) in that version (see https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3719), as documented [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/extensibility/payment-method-integration.md#client-side-integration) (though only under "Express payment methods"?)

**Comment by dechov** [comment]
> (According to [this doc](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/extensibility/payment-method-integration.md#server-side-integration), it looks like maybe we should be accessing that data differently: `wc.wcSettings.getSetting( 'woocommerce_payments_data' )`?)

Oh wait, we are doing that [[source](https://github.com/Automattic/woocommerce-payments/blob/be5c9193730ba603f03b525e9663fe1d80f96db3/client/utils/checkout.js#L13)], but it does first assume `wcpay_config` is _defined_ – that must be the issue.

**Comment by RadoslavGeorgiev** [comment]
> Oh wait, we are doing that [source], but it does first assume `wcpay_config` is _defined_ – that must be the issue.

Yes, it seems to be the issue. It used to be like this:

```js
const config = ( 'undefined' !== typeof wcpay_config )
	// eslint-disable-next-line camelcase
	? wcpay_config // Classic checkout
	: wc.wcSettings.getSetting( 'woocommerce_payments_data' ); // Blocks

return config[ name ] || null;
```

But in https://github.com/Automattic/woocommerce-payments/commit/d8d7cf8582194180a538d0cd6687240d1281cc76#diff-a40db4c408c37b4ca3bcc70dc6c65fdd454844df00b32b7048243ab9d87a68f5 it got transformed to the new format, which does not check whether `wcpay_config` is defined or not.

**Comment by jrodger** [comment]
> With WooCommerce Payments 1.8.0 or earlier but with WooCommerce Blocks 4.4.0 (released Feb 3), a different JS error is thrown: Uncaught Error: The features property for the payment method must be an array. – this is because the payment method requires a new features array (inside a supports object property) in that version (see woocommerce/woocommerce-gutenberg-products-block#3719), as documented here (though only under "Express payment methods"?)

This is good to know, but I don't think there is anything we can do in WCPay right? A merchant would either need to upgrade to WCPay 1.9.2, or upgrade to a version of WooCommerce Blocks that fixes the issue (assuming that's possible)?

**Comment by dechov** [comment]
@jrodger Right – nothing we can do in WCPay besides release 1.9.2 - it will be addressed in the Blocks extension's next release as well (https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3796).

**Comment by dechov** [comment]
Fixed in #1285 and #1290
